Mathlib/Probability/Kernel/MeasureCompProd.lean

```diff
@@ -118,4 +118,28 @@ instance [IsFiniteMeasure μ] [IsFiniteKernel κ] : IsFiniteMeasure (μ ⊗ₘ 
 instance [IsProbabilityMeasure μ] [IsMarkovKernel κ] : IsProbabilityMeasure (μ ⊗ₘ κ) := by
   rw [compProd]; infer_instance
 
+lemma absolutelyContinuous_compProd_left {ν : Measure α} [SFinite μ] [SFinite ν]
```

Comment:
You shouldn't need `SFinite μ` here, as a measure which is absolutely continuous wrt an s-finite measure is itself s-finite, right? (Not sure we have that in mathlib though).

Reply:
I just proved that and will PR it. The way I did it ended up needing an unexpectedly large amount of stuff about s-finite measures, so that will be several PRs containing the following:
- if a measure `μ` is s-finite, then `μ.withDensity f` is s-finite
- any s-finite measure can be written as `withDensity` of a finite measure
- if we have two measures `μ, ν` and `ν` is finite, then `μ` can be written as a sigma-finite measure plus a measure which is infinite when `ν` is not zero. Halmos calls the proof technique used here the method of exhaustion.
- putting all of that together: 
```
lemma sFinite_of_absolutelyContinuous {μ ν : Measure α} [SFinite ν] (hμν : μ ≪ ν) : SFinite μ
```

That material should also allow us to generalize our Radon-Nikodym theorem.

Reply:
That's great!